lib/python/ray/graph.py

```diff
@@ -0,0 +1,58 @@
+# Utilities to deal with computation graphs
+
+import os
+import datetime
+import ray.internal.graph_pb2
+import graphviz
+
+def graph_to_dot(computation_graph, out_filepath=None, view=False):
+  """
+  Write rendered computation graph to file and optionally display it on screen.
+
+  Args:
+    computation_graph [graph_pb2.CompGraph]: protocol buffer description of
+      the computation graph
+
+    out_filepath: Path where the dot file of the computation graph will be stored.
+      The graph will be rendered and saved in out_filepath.pdf
+
+    view: If true, the result the python graphviz package will try to open the
+      result in a viewer
+
+  Example:
+    In ray/scripts, call "python shell.py" and paste in the following code.
+    This will create a file linalg.out with the .dot code and a file linalg.out.pdf
+    with the rendered computation graph.
+
+    x = da.zeros([20, 20])
+    y = da.zeros([20, 20])
+    z = da.dot(x, y)
+
+    ray.worker.dump_computation_graph("linalg.binaryproto")
+
+    g = ray.internal.graph_pb2.CompGraph()
+    g.ParseFromString(open("linalg.binaryproto").read())
+
+    ray.graph_to_dot(g, "linalg.out")
+  """
+  dot = graphviz.Digraph(format="pdf")
+  for (i, op) in enumerate(computation_graph.operation):
+    if op.HasField("task"):
+      dot.node("op" + str(i), shape="box", label=str(i) + "\n" + op.task.name.split(".")[-1])
+      for res in op.task.result:
+        dot.edge("op" + str(i), str(res))
+    elif op.HasField("put"):
+      dot.node("op" + str(i), shape="box", label=str(i) + "\n" + "put")
+      dot.edge("op" + str(i), str(op.put.objref))
+    elif op.HasField("get"):
+      dot.node("op" + str(i), shape="box", label=str(i) + "\n" + "get")
+    creator_operationid = op.creator_operationid if op.creator_operationid != 2 ** 64 - 1 else "-root"
+    dot.edge("op" + str(creator_operationid), "op" + str(i), style="dotted", constraint="false")
+    for arg in op.task.arg:
+      if not arg.HasField("obj"):
+        dot.node(str(arg.ref))
+        dot.edge(str(arg.ref), "op" + str(i))
+  if out_filepath == None:
+    out_filepath = os.path.join(ray.config.LOG_DIRECTORY, (ray.config.LOG_TIMESTAMP + "-computation-graph.out").format(datetime.datetime.now()))
+  print "Writing computation graph to file {}.pdf".format(out_filepath)
```

Comment:
On my machine, the command `ray.graph_to_dot(g, "linalg.out")` writes to `linalg.out`, not `linalg.out.pdf`.